server/portal/apps/intromessages/views.py

```diff
@@ -0,0 +1,35 @@
+"""
+.. :module: apps.intromessages.views
+   :synopsis: Views to handle read/unread status of IntroMessages
+"""
+
+import logging
+from portal.views.base import BaseApiView
+from django.http import JsonResponse
+from portal.apps.intromessages.models import IntroMessages
+from django.contrib.auth.decorators import login_required
+from django.utils.decorators import method_decorator
+import json
+
+
+logger = logging.getLogger(__name__)
+
+
+@method_decorator(login_required, name='dispatch')
+class IntroMessagesView(BaseApiView):
+    def get(self, request, *args, **kwargs):
+        messages_array = IntroMessages.objects.filter(user=request.user).values('component', 'unread')
+        return JsonResponse({'response': list(messages_array)})
+
+    def put(self, request, *args):
+        body = json.loads(request.body)
+        for component_name, component_value in body.items():
+            db_message = IntroMessages.objects.filter(user=request.user, component=component_name)
```

Comment:
Instead of `filter`, could we use `get`? That way the result is already an object, and we don't have to worry about dealing with the possibility of an array of values with a length longer than 1.

If we go with `objects.get`, we'll want to wrap in a try/except block

Reply:
Sorry.  I should have figured that out myself.  Changes are complete.